tests/rule_based_profiler/alpha/test_profiler_user_workflows.py

```diff
@@ -306,3 +307,99 @@ def test_bobster_profiler_user_workflow_multi_batch_row_count_range_rule_bootstr
             "test_configuration_bootstrap_sampling_method"
         ]["expect_table_row_count_to_be_between_max_value_mean_value"]
     )
+
+
+@pytest.mark.skipif(
+    version.parse(np.version.version) < version.parse("1.21.0"),
+    reason="requires numpy version 1.21.0 or newer",
+)
+def test_quentin_profiler_user_workflow_multi_batch_quantiles_value_ranges_rule(
+    quentin_columnar_table_multi_batch_data_context,
+    quentin_columnar_table_multi_batch,
+):
+    # Load data context
+    data_context: DataContext = quentin_columnar_table_multi_batch_data_context
+
+    # Load profiler configs & loop (run tests for each one)
+    yaml_config: str = quentin_columnar_table_multi_batch["profiler_config"]
+
+    # Instantiate Profiler
+    profiler_config: CommentedMap = yaml.load(yaml_config)
+
+    # Roundtrip through schema validation to remove any illegal fields add/or restore any missing fields.
+    deserialized_config: dict = ruleBasedProfilerConfigSchema.load(profiler_config)
+    serialized_config: dict = ruleBasedProfilerConfigSchema.dump(deserialized_config)
+
+    # `class_name`/`module_name` are generally consumed through `instantiate_class_from_config`
+    # so we need to manually remove those values if we wish to use the **kwargs instantiation pattern
+    serialized_config.pop("class_name")
+    serialized_config.pop("module_name")
+
+    profiler: RuleBasedProfiler = RuleBasedProfiler(
+        **serialized_config,
+        data_context=data_context,
+    )
+
+    expectation_suite: ExpectationSuite = profiler.run(
+        expectation_suite_name=quentin_columnar_table_multi_batch["test_configuration"][
+            "expectation_suite_name"
+        ],
+    )
+
+    expectation_configuration_dict: dict
+    column_name: str
+    expectation_kwargs: dict
+    expect_column_quantile_values_to_be_between_expectation_configurations_kwargs_dict: Dict[
+        str, dict
+    ] = {
+        expectation_configuration_dict["kwargs"][
+            "column"
+        ]: expectation_configuration_dict["kwargs"]
+        for expectation_configuration_dict in expectation_suite.to_json_dict()[
+            "expectations"
+        ]
+    }
+    expect_column_quantile_values_to_be_between_expectation_configurations_value_ranges_by_column: Dict[
+        str, List[List[Number]]
+    ] = {
+        column_name: expectation_kwargs["value_ranges"]
+        for column_name, expectation_kwargs in expect_column_quantile_values_to_be_between_expectation_configurations_kwargs_dict.items()
+    }
```

Comment:
Nit - This is all really difficult to read in my opinion. I love comprehensions but I think they shouldn't be used in places where a loop is more readable.

I'd ask that these be refactored but won't block the PR on this.

Reply:
🙇 -- I can explain why they are a natural for it (basically, extracting relevant aspects of our rich `ExpectationConfiguration` parameters so as to aid in assertions more easily).